vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -228,115 +205,97 @@ bool ChakraRuntime::isInspectable() {
 }
 
 facebook::jsi::Runtime::PointerValue *ChakraRuntime::cloneSymbol(
-    const facebook::jsi::Runtime::PointerValue *) {
-  throw facebook::jsi::JSINativeException("Not implemented!");
+    const facebook::jsi::Runtime::PointerValue *pointerValue) {
+  return CloneChakraPointerValue(pointerValue);
 }
 
 facebook::jsi::Runtime::PointerValue *ChakraRuntime::cloneString(
-    const facebook::jsi::Runtime::PointerValue *pv) {
-  if (!pv) {
-    return nullptr;
-  }
-  const ChakraStringValue *string = static_cast<const ChakraStringValue *>(pv);
-  return makeStringValue(string->m_str);
+    const facebook::jsi::Runtime::PointerValue *pointerValue) {
+  return CloneChakraPointerValue(pointerValue);
 }
 
 facebook::jsi::Runtime::PointerValue *ChakraRuntime::cloneObject(
-    const facebook::jsi::Runtime::PointerValue *pv) {
-  if (!pv) {
-    return nullptr;
-  }
-
-  const ChakraObjectValue *object = static_cast<const ChakraObjectValue *>(pv);
-  return makeObjectValue(object->m_obj);
+    const facebook::jsi::Runtime::PointerValue *pointerValue) {
+  return CloneChakraPointerValue(pointerValue);
 }
 
 facebook::jsi::Runtime::PointerValue *ChakraRuntime::clonePropNameID(
-    const facebook::jsi::Runtime::PointerValue *pv) {
-  if (!pv) {
-    return nullptr;
-  }
-
-  const ChakraPropertyIdValue *propId =
-      static_cast<const ChakraPropertyIdValue *>(pv);
-  return makePropertyIdValue(propId->m_propId);
+    const facebook::jsi::Runtime::PointerValue *pointerValue) {
+  return CloneChakraPointerValue(pointerValue);
 }
 
 facebook::jsi::PropNameID ChakraRuntime::createPropNameIDFromAscii(
     const char *str,
     size_t length) {
-  JsValueRef propIdRef = createJSPropertyId(str, length);
-
-  auto res = createPropNameID(propIdRef);
-  return res;
+  // Unfortunately due to the typing used by JSI and Chakra(Core), we have to do
+  // a double reinterpret cast here.
```

Comment:
>double reinterpret cast here [](start = 7, length = 28)

It may be more readable to have the overload below cast to char* and call this one instead

Reply:
Wouldn't it be odd semantically for the UTF8 version to call the Ascii version?

---
In reply to: [336266662](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336266662) [](ancestors = 336266662)